### PR TITLE
shell: Touch support for move and resize grabs

### PR DESCRIPTION
**Pull request by ids1024**
Moving Gnome Terminal with touch now works, except the window is shown frozen in place even as the screen indicates tiling or a stack being created, which happens on release?

Also need to add touch to other grabs, and have SSDs (and libcosmic) start move/etc. based on touch events.

### CODE REVIEW

**Comment by ids1024** [comment]
Move grabs are working pretty nicely with touch now. And this has resize grabs.

Though I see I've broken the rendering for moving on other monitors. And resizing with `ResizeSurfaceGrab`  breaks moving between monitors.

**Comment by ids1024** [comment]
https://github.com/pop-os/cosmic-comp/pull/415 fixes one of those issues. Added a commit fixing the issue moving between monitors.

There does seem to be another issue when the pointer isn't on the same output as touch events.

File: src/shell/focus/target.rs [hunk 1]
```diff
@@ -497,8 +497,8 @@ impl TouchTarget<State> for PointerFocusTarget {
                 TouchTarget::down(window, seat, data, event, seq)
             }
             PointerFocusTarget::StackUI(stack) => TouchTarget::down(stack, seat, data, event, seq),
-            // TODO: implement TouchTarget for ResizeFork/Grabs
-            PointerFocusTarget::ResizeFork(_fork) => {}
+            // TODO: implement TouchTarget for Grabs
```
Comment (Drakulix) [inline src/shell/focus/target.rs:500]: Is this comment still relevant with the latest changes?

**Review by Drakulix — APPROVED** [review]
LGTM!

File: src/shell/focus/target.rs (hunk 1 above)
Comment (ids1024) [reply src/shell/focus/target.rs:500]: Now that all the grabs should have touch support, I guess that comment isn't really applicable anymore.

Though handling touch events on SSDs to start drags is still needed.

Comment (Drakulix) [reply src/shell/focus/target.rs]: > Though handling touch events on SSDs to start drags is still needed.

Is that something you want to address before I go ahead and merge this?

Comment (ids1024) [reply src/shell/focus/target.rs]: It's probably not that hard to add, but it can be done later.

Comment (ids1024) [reply src/shell/focus/target.rs]: My cosmic-comp changes for https://github.com/Smithay/smithay/pull/1386 are also based on this PR.

**Pull request merged**